fem/src/CMakeLists.txt

```diff
@@ -172,11 +172,15 @@ IF(HAVE_PARMMG)
 ENDIF()
   
 IF(HAVE_ZOLTAN)
-  TARGET_LINK_LIBRARIES(elmersolver ${ZOLTAN_LIBRARY})
   IF(Zoltan_ENABLE_ParMETIS)
     TARGET_LINK_LIBRARIES(elmersolver ${ParMetis_LIBRARIES})
   ENDIF()
```

Comment:
Not sure about this. But the existing structure looks like the Zoltan library might depend on the ParMetis library.
For some linkers (e.g., `ld.bfd`) the order of linker flags matters.
If you move linking to `${ZOLTAN_LIBRARY}` down, it might be safest to keep this (conditional) block after that.

Reply:
Or keep this block here but move linking to the Zoltan libraries (conditionally via `${ZOLTAN_LIBRARY}` or `Zoltan::zoltan`) back up to where it was.

Reply:
Did you mean something like this:
```
IF(HAVE_ZOLTAN)
  IF(USE_SYSTEM_ZOLTAN)
    TARGET_LINK_LIBRARIES(elmersolver Zoltan::zoltan)
  ELSE()
    ADD_DEPENDENCIES(elmersolver zoltan)
    TARGET_LINK_LIBRARIES(elmersolver ${ZOLTAN_LIBRARY})
  ENDIF()
  IF(Zoltan_ENABLE_ParMETIS)
    TARGET_LINK_LIBRARIES(elmersolver ${ParMetis_LIBRARIES})
  ENDIF()
ENDIF()
```

Reply:
Yes. That looks good to me. 👍